openedx_learning/core/contents/api.py

```
@@ -41,37 +53,49 @@ def create_text_from_raw_content(raw_content, encoding="utf-8-sig"):
 
 
 def get_or_create_raw_content(
-    learning_package_id, data_bytes, mime_type, created, hash_digest=None
-):
+    learning_package_id: int,
+    data_bytes: bytes,
+    mime_type: str,
+    created: datetime,
+    hash_digest: str | None = None,
+) -> tuple[RawContent, bool]:
+    """
+    Get the RawContent in the given learning package with the specified data,
+    or create it if it doesn't exist.
+    """
     hash_digest = hash_digest or create_hash_digest(data_bytes)
     try:
         raw_content = RawContent.objects.get(
             learning_package_id=learning_package_id, hash_digest=hash_digest
         )
-        created = False
+        was_created = False
     except RawContent.DoesNotExist:
         raw_content = create_raw_content(
             learning_package_id, data_bytes, mime_type, created, hash_digest
         )
-        created = True
+        was_created = True
 
-    return raw_content, created
+    return raw_content, was_created
 
 
 def get_or_create_text_content_from_bytes(
-    learning_package_id,
-    data_bytes,
-    mime_type,
-    created,
-    hash_digest=None,
-    encoding="utf-8-sig",
+    learning_package_id: int,
+    data_bytes: bytes,
+    mime_type: str,
+    created: datetime,
+    hash_digest: str | None = None,
+    encoding: str = "utf-8-sig",
 ):
+    """
+    Get the TextContent in the given learning package with the specified data,
+    or create it if it doesn't exist.
+    """
     with atomic():
         raw_content, rc_created = get_or_create_raw_content(
-            learning_package_id, data_bytes, mime_type, created, hash_digest=None
+            learning_package_id, data_bytes, mime_type, created, hash_digest
         )
         if rc_created or not hasattr(raw_content, "text_content"):
-            text = codecs.decode(data_bytes, "utf-8-sig")
+            text = codecs.decode(data_bytes, encoding)
```

Same here: `encoding` parameter was silently ignored.